### ISSUE TITLE
[Bug] mongodb doesn't support authenticationDatabase

### ISSUE THREAD

**Issue by wei-hai** [influxdata/telegraf#3347]
### Relevant telegraf.conf:
```
[[inputs.mongodb]]
  ## An array of URLs of the form:
  ##   "mongodb://" [user ":" pass "@"] host [ ":" port]
  ## For example:
  ##   mongodb://user:auth_key@10.10.3.30:27017,
  ##   mongodb://10.10.3.33:18832,
  servers = ["mongodb://127.0.0.1:27017"]
  gather_perdb_stats = true

  ## Optional SSL Config
  # ssl_ca = "/etc/telegraf/ca.pem"
  # ssl_cert = "/etc/telegraf/cert.pem"
  # ssl_key = "/etc/telegraf/key.pem"
  ## Use SSL but skip chain & host verification
  # insecure_skip_verify = false
```
### System info:
ubuntu 14
[Include Telegraf version, operating system name, and other relevant details]
1.4.2

By default the authenticationDatabase is `admin`, the plugin should support customized authenticationDatabase in the url like this: `mongodb://user:auth_key@10.10.3.30:27017/authenticationDatabase`

**Comment by danielnelson** [comment]
It makes sense to support more of the connection string.  Here is some documentation I found about the [connection string format](https://docs.mongodb.com/manual/reference/connection-string/).

**Comment by Ayrdrie** [comment]
We looked into this issue and don’t believe this is a connection problem with the plugin.  We set up a user to authenticate to another database other than “admin”.  The connection was successful but the plugin generated

We looked into the issue and it looks like this isn’t a connection issue but rather a Mongo permissions problem.  The plugin executes the serverStatus command and Mongo requires the user to have “Admin” privileges in the “admin” database.  We believe the error you are seeing was because the user didn’t have the correct permissions that are required by the plugin to run.  Let us know if we are misinterpreting the issue you are describing. 

Here are the commands we used to setup the user for the plugin:  

`db.createUser({user: "telegraf", pwd: "telegraf", roles:[{role:"userAdminAnyDatabase",db:"admin"}]})`

This user was able to connect with the following telegraf.conf connection string.

`servers = ["mongodb://telegraf:telegraf@<host_address>:27017/telegraf"]`

**Comment by Ayrdrie** [comment]
We looked into the issue and it looks like this isn’t a connection issue but rather a Mongo permissions problem.  The plugin executes the serverStatus command and Mongo requires the user to have “Admin” privileges in the “admin” database.  We believe the error you are seeing was because the user didn’t have the correct permissions that are required by the plugin to run.  Let us know if we are misinterpreting the issue you are describing. 

Here are the commands we used to setup the user for the plugin:  

`db.createUser({user: "telegraf", pwd: "telegraf", roles: [{role:"userAdminAnyDatabase",db:"admin"}]})`

This user was able to connect with the following telegraf.conf connection string.

`servers = ["mongodb://telegraf:telegraf@<host_address>:27017/telegraf"]`
